spec/rubocop/cop/rspec/empty_hook_spec.rb

```diff
@@ -0,0 +1,418 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::EmptyHook do
+  subject(:cop) { described_class.new }
+
+  context 'with `before` hook' do
+    it 'detects offense for empty `before`' do
+      expect_offense(<<~RUBY)
+        before {}
+        ^^^^^^ Empty hook detected.
+      RUBY
+    end
+
+    it 'detects offense for empty `before` with :each scope' do
+      expect_offense(<<~RUBY)
+        before(:each) {}
+        ^^^^^^^^^^^^^ Empty hook detected.
+      RUBY
+    end
+
+    it 'detects offense for empty `before` with :example scope' do
+      expect_offense(<<~RUBY)
+        before(:example) {}
+        ^^^^^^^^^^^^^^^^ Empty hook detected.
+      RUBY
+    end
+
+    it 'detects offense for empty `before` with :context scope' do
+      expect_offense(<<~RUBY)
+        before(:context) {}
+        ^^^^^^^^^^^^^^^^ Empty hook detected.
+      RUBY
+    end
+
+    it 'detects offense for empty `before` with :all scope' do
+      expect_offense(<<~RUBY)
+        before(:all) {}
+        ^^^^^^^^^^^^ Empty hook detected.
+      RUBY
+    end
+
+    it 'detects offense for empty `before` with :suite scope' do
+      expect_offense(<<~RUBY)
+        before(:suite) {}
+        ^^^^^^^^^^^^^^ Empty hook detected.
+      RUBY
+    end
+
+    it 'accepts non-empty `before` hook' do
+      expect_no_offenses(<<~RUBY)
+        before { create_users }
+      RUBY
+    end
+
+    it 'accepts multiline `before` hook' do
+      expect_no_offenses(<<~RUBY)
+        before(:all) do
+          create_users
+          create_products
+        end
+      RUBY
+    end
+  end
+
+  context 'with `after` hook' do
+    it 'detects offense for empty `after`' do
+      expect_offense(<<~RUBY)
+        after {}
+        ^^^^^ Empty hook detected.
+      RUBY
+    end
+
+    it 'detects offense for empty `after` with :each scope' do
+      expect_offense(<<~RUBY)
+        after(:each) {}
+        ^^^^^^^^^^^^ Empty hook detected.
+      RUBY
+    end
```

Comment:
I love how you cover every possible case. However, with N hooks and M scopes, it's getting quite unwieldy.
Would you agree to keep only a few distinctive examples for each hook type to keep this spec short?

E.g. all scopes for `before`, `after {}`, `around {}`, `around { yield }`, `around { setup; yield }`, `prepend_before {}`, `append_before {}`, `prepend_after {}`, `append_after {}`, and maybe keep "accepts" examples for `before`/`prepend_before`/`append_before`/`prepend_after`/`append_after` if you see fit.

I can't remember a good example when the lack of covering the full matrix has missed a real problem given a simple combination of matchers in the node pattern.

In any case, `around(:suite)` is illegal as to my best knowledge, and `around(:context)`/`around(:all)` issues a warning that it actually behaves like `around(:example)`.

Reply:
PS I really love how semantic current examples are, e.g. `before { setup_users }` vs `after { cleanup_users` 👍

Reply:
I agree the specs are a bit long and maybe too explicit.

One way to shorten them might be to expect more than one type of offense at the same time, e.g.:

```rb
    it 'detects offenses for empty `after` hooks' do
      expect_offense(<<~RUBY)
        after {}
        ^^^^^ Empty hook detected.
        after(:each) {}
        ^^^^^^^^^^^^ Empty hook detected.
        after(:example) {}
        ^^^^^^^^^^^^^^^ Empty hook detected.
        after(:context) {}
        ^^^^^^^^^^^^^^^ Empty hook detected.
        after(:all) {}
        ^^^^^^^^^^^ Empty hook detected.
        after(:suite) {}
        ^^^^^^^^^^^^^ Empty hook detected.
      RUBY
    end
```